grpc/interceptors.go

```diff
@@ -18,17 +18,38 @@ type serverInterceptor struct {
 
 func (si *serverInterceptor) intercept(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	if info == nil {
-		si.stats.Inc("gRPC.NoInfo", 1)
+		si.stats.Inc("gRPCServer.NoInfo", 1)
 		return nil, errors.New("passed nil *grpc.UnaryServerInfo")
 	}
-	si.stats.Inc(fmt.Sprintf("gRPC.%s", info.FullMethod), 1)
-	si.stats.GaugeDelta(fmt.Sprintf("gRPC.%s.InProgress", info.FullMethod), 1)
 	s := si.clk.Now()
+	si.stats.Inc(fmt.Sprintf("gRPCServer.%s", info.FullMethod), 1)
+	si.stats.GaugeDelta(fmt.Sprintf("gRPCServer.%s.InProgress", info.FullMethod), 1)
 	resp, err := handler(ctx, req)
-	si.stats.TimingDuration(fmt.Sprintf("gRPC.%s", info.FullMethod), si.clk.Now().Sub(s))
-	si.stats.GaugeDelta(fmt.Sprintf("gRPC.%s.InProgress", info.FullMethod), -1)
+	si.stats.TimingDuration(fmt.Sprintf("gRPCServer.%s", info.FullMethod), si.clk.Since(s))
+	si.stats.GaugeDelta(fmt.Sprintf("gRPCServer.%s.InProgress", info.FullMethod), -1)
 	if err != nil {
-		si.stats.Inc(fmt.Sprintf("gRPC.%s.Failed", info.FullMethod), 1)
+		si.stats.Inc(fmt.Sprintf("gRPCServer.%s.Failed", info.FullMethod), 1)
 	}
 	return resp, err
 }
+
+type clientInterceptor struct {
+	stats metrics.Scope
+	clk   clock.Clock
+}
+
+// intercept fulfils the grpc.UnaryClientInterceptor interface, it should be noted that while this API
+// is currently experimental the metrics it reports should be kept as stable as can be, *within reason*
+func (ci *clientInterceptor) intercept(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	s := ci.clk.Now()
+	ci.stats.Inc(fmt.Sprintf("gRPCClient.%s", method), 1)
```

Comment:
Seems like this method would be a good use for `ci.stats.Scope("gRPCClient", method)`.